### PR TITLE
Fix specs

### PR DESCRIPTION
**Pull request by jedmao**

### CODE REVIEW

**Comment by jedmao** [comment]
@ooHmartY @ethanselzer @Intregrisist @BradLook please review.

File: events.spec.js [hunk 1]
```diff
@@ -78,13 +92,13 @@ describe("Event Dispatcher", function() {
...
+        events.on('foo', adder);
+        events.trigger('foo', 1);
         expect(bar).toBe(2);
 
-        Events.on('foo', multiplier)
-        Events.off('foo', adder);
-        Events.trigger('foo', 100);
+        events.on('foo', multiplier)
+        events.off('foo', adder);
+        events.trigger('foo', 100);
```
Comment (ethanselzer) [inline events.spec.js]: In isolation, I'm minorly concerned that this test could return a false positive. Here is the scenario:
1 `multiplier` actually overwrites `adder` on line 99
2 `adder` is not found on line 100
3 `multiplier` is the only listener on line 101

I realize that this scenario is guarded against accross the whole suite. My comment is strictly about this test in isolation.

I propose the following:
```
events.on('foo', multiplier);
events.on('foo', adder);
events.off('foo', adder);
trigger('foo', 100)
expect(bar).toBe(100);
```

File: events.spec.js [hunk 2]
```diff
@@ -1,73 +1,87 @@
...
+    it('is a function', function() {
+        expect(typeof Events).toBe('function');
     });
 
-    xit("can register a callback", function() {
-        Events.on('foo', function() {
-            return 'bar';
-        });
+    xit('can register a callback', function() {
+        var events = new Events();
```
Comment (ethanselzer) [inline events.spec.js]: I'm guessing you considered using `beforeEach`. How come you decided against it?

File: events.spec.js [hunk 3]
```diff
@@ -78,13 +92,13 @@ describe("Event Dispatcher", function() {
...
-        Events.on('foo', adder);
-        Events.trigger('foo', 1);
+        events.on('foo', adder);
+        events.trigger('foo', 1);
         expect(bar).toBe(2);
 
-        Events.on('foo', multiplier)
-        Events.off('foo', adder);
-        Events.trigger('foo', 100);
+        events.on('foo', multiplier)
```
Comment (ethanselzer) [inline events.spec.js]: Missing semicolon. Maybe we should add a linter. The clever candidate would look at package.json, see it, and lint their work!

File: events.spec.js (hunk 2 above)
Comment (jedmao) [inline events.spec.js]: I was having issues with listeners from a previous spec being triggered in the next spec. `beforeEach` wasn't fixing the issue, but it seems to be working fine right now, so I'll reintroduce it.

**Comment by ethanselzer** [comment]
Thanks for doing this Jed!

File: events.spec.js (hunk 1 above)
Comment (jedmao) [inline events.spec.js]: Thanks. I'm going to create some jasmine spies to avoid this whole conundrum altogether.

File: events.spec.js (hunk 3 above)
Comment (jedmao) [inline events.spec.js]: I'll add eslint. I wanted to anyway. Thanks.

**Comment by jedmao** [comment]
@ethanselzer please review the last 2 commits and let me know if you want to add any more ESLint rules. @Intregrisist, you'll be happy to know I used Jasmine spies for most specs.

**Comment by jedmao** [comment]
@ethanselzer see 8c03355.

**Comment by ethanselzer** [comment]
:+1:

**Pull request merged**